### PR TITLE
support customized base for fused_rope

### PR DESCRIPTION
**Pull request by deepllz**
### PR Category

Others

### PR Types

Others

### Description

fused_rope支持自定义base
测试发现将rotary_base通过参数传入到kernel会导致cuda kernel降速，所以对默认的10000 base，特化模板，保证base=10000的时候，kernel不降速。
Pcard-73145

### CODE REVIEW

**Review by heavyrain-lzy — DISMISSED** [review]
LGTM for YAML

**Review by LiYuRio — DISMISSED** [review]

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM，
中文文档也同步更新一下 https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/incubate/nn/functional/fused_rotary_position_embedding_cn.html

**Review by jeff41404 — DISMISSED** [review]
LGTM for API

**Review by yuanlehome — DISMISSED** [review]

**Review by vivienfanghuagood — APPROVED** [review]

**Review by heavengate — DISMISSED** [review]

File: paddle/phi/kernels/fusion/gpu/fused_rope_utils.h [hunk 1]
```diff
@@ -115,58 +257,86 @@ __global__ void VectorizedFusedRopeWithRotateEveryTwoKernel(
   int64_t size = batch_size * seq_len * num_heads * head_dim;
   MPType sin_value[VecSize];
   MPType cos_value[VecSize];
+
+  if (static_cast<int>(rotary_emb_base) == kDefaultRotaryBase) {
```
Comment (sneaxiy) [inline paddle/phi/kernels/fusion/gpu/fused_rope_utils.h]: Why cast to `int`? What if `rotary_emb_base = 10000.1`?

Should be something like `if (rotary_emb_base == static_cast<decltype(rotary_emb_base)>(kDefaultRotaryBase)`.

File: paddle/phi/kernels/fusion/gpu/fused_rope_utils.h [hunk 2]
```diff
@@ -195,52 +366,56 @@ __global__ void VectorizedFusedRopeWithRotateHalfKernel(
...
-      for (int nx = 0; nx < VecSize; ++nx) {
-        MPType p0 = static_cast<MPType>(input_v[nx]);
-        MPType p1 = static_cast<MPType>(input_r[nx]);
-
-        result[nx] = cos_value[nx] * p0 + sign * sign_r * sin_value[nx] * p1;
-
-        store[nx] = static_cast<T>(result[nx]);
-      }
-      out[0] = *(reinterpret_cast<VecType*>(store));
+  if (static_cast<int>(rotary_emb_base) == kDefaultRotaryBase) {
```
Comment (sneaxiy) [inline paddle/phi/kernels/fusion/gpu/fused_rope_utils.h]: Same as above.

**Review by sneaxiy — COMMENTED** [review]
Almost LGTM except some changes

File: paddle/phi/kernels/fusion/gpu/fused_rope_utils.h (hunk 1 above)
Comment (deepllz) [reply paddle/phi/kernels/fusion/gpu/fused_rope_utils.h]: Done

File: paddle/phi/kernels/fusion/gpu/fused_rope_utils.h (hunk 2 above)
Comment (deepllz) [reply paddle/phi/kernels/fusion/gpu/fused_rope_utils.h]: Done

**Review by sneaxiy — APPROVED** [review]

**Review by LiYuRio — APPROVED** [review]

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Review by jeff41404 — APPROVED** [review]

**Review by heavengate — APPROVED** [review]

**Pull request merged**